src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -1581,6 +1587,40 @@ private static TempPaths GetTempPaths()
             }
         }
 
+        internal class TemporaryEnvironment : IDisposable
+        {
+            private bool _disposed;
+            private readonly string _name;
+            private readonly string _originalValue;
+
+            public TemporaryEnvironment(string name, string value)
```

Comment:
I envisioned a cascading set of usings but yeah, that's more elegant. Mind if we wait until after RC2?